src/Compiler/lex.fsl

```diff
@@ -1427,15 +1455,124 @@ and tripleQuoteString sargs skip = parse
+          // Add two % chars for each % that is supposed to be treated as regular string content
+          // + 1 for a format specifier.
+          let percentsToEmit =
+              if numPercents < args.delimLength then 2 * numPercents
+              else 2 * (numPercents - args.delimLength) + 1
+          let s = String.replicate percentsToEmit "%"
+          addUnicodeString buf s
+          result() }
+
+ | "{" +
+    { let (buf, fin, m, kind, args) = sargs
+      let numBraces = String.length (lexeme lexbuf)
+      // Extended interpolated strings starts with at least 2 $
+      // Number of leading $s is the number of '{' needed to open interpolation expression (delimLength)
+      // 2x delimLength (or more) of '{' in a row would be unambiguous, so it's disallowed
+      let maxBraces = 2 * args.delimLength - 1
+      if numBraces > maxBraces then
+          let m2 = lexbuf.LexemeRange
+          args.stringNest <- (1, LexerStringStyle.ExtendedInterpolated, args.delimLength, m2) :: args.stringNest
+          let cont = LexCont.Token(args.ifdefStack, args.stringNest)
+          fail args lexbuf
+              (FSComp.SR.lexTooManyLBracesInTripleQuote())
+              (fin.Finish buf kind (LexerStringFinisherContext.InterpolatedPart ||| LexerStringFinisherContext.TripleQuote) cont)
+      elif numBraces < args.delimLength then
+          // Less than delimLength means we treat '{' as normal content
+          addUnicodeString buf (lexeme lexbuf)
+          if not skip then STRING_TEXT (LexCont.String(args.ifdefStack, args.stringNest, LexerStringStyle.ExtendedInterpolated, kind, args.delimLength, m))
+          else extendedInterpolatedString sargs skip lexbuf
+      // numBraces in [delimLength; maxBraces)
+      else
+          // A sequence of delimLength * '{' starts interpolation expression.
+          // Any extra '{' are treated as normal string content.
+          let extraBraces = numBraces - args.delimLength
+          if extraBraces > 0 then
+              String.replicate extraBraces "{" |> addUnicodeString buf
+          // get a new range for where the fill starts
+          let m2 = lexbuf.LexemeRange
+          args.stringNest <- (1, LexerStringStyle.ExtendedInterpolated, args.delimLength, m2) :: args.stringNest
+          let cont = LexCont.Token(args.ifdefStack, args.stringNest)
+          fin.Finish buf kind (LexerStringFinisherContext.InterpolatedPart ||| LexerStringFinisherContext.TripleQuote) cont
+    }
+
+ | "}" +
+    { let (buf, _fin, m, kind, args) = sargs
+      let numBraces = lexeme lexbuf |> String.length
+      let result() =
+          if not skip then STRING_TEXT (LexCont.String(args.ifdefStack, args.stringNest, LexerStringStyle.ExtendedInterpolated, kind, args.delimLength, m))
+          else extendedInterpolatedString sargs skip lexbuf
+      if args.delimLength > numBraces then
+          lexeme lexbuf |> addUnicodeString buf
+          (result())
+      else // numBraces >= args.delimLength
```

Comment:
What is wrong with the equal scenario?
numBraces = args.delimLength

Or are "numBraces" actually "number of additional braces" on top of the 1 we are currently matching?

Reply:
It's the number of additional braces, in the sense that, if they are directly adjacent to the interpolation expression, these are the braces left over after those that close the interpolation.

To give an example:
```fs
// these were matched within the `token` rule -vvv
let s1 = $$$""" six times seven equals {{{ 6*7 }}}}}} """
// we only see these "extra braces" here ---------^^^
```

(and if they are not adjacent to the interpolation expression, we don't allow that either, there can be n-1 in content of the string, n is used for delimiting the interpolation)